File: ios/DebugModule.swift
import BitmovinPlayer
import ExpoModulesCore

public class DebugModule: Module {
    public func definition() -> ModuleDefinition {
        Name("DebugModule")

        AsyncFunction("setDebugLoggingEnabled") { (enabled: Bool) in
            DebugConfig.logging.logger?.level = enabled ? .verbose : .warning
        }.runOnQueue(.main)
    }
}

File: android/src/main/java/com/bitmovin/player/reactnative/DebugModule.kt
package com.bitmovin.player.reactnative

import com.bitmovin.player.api.DebugConfig
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DebugModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("DebugModule")

        AsyncFunction("setDebugLoggingEnabled") { enabled: Boolean ->
            DebugConfig.isLoggingEnabled = enabled
        }
    }
}
